### PR TITLE
AVX-22266 [FireNet Data source] Causes crash in provider when running…

### PR DESCRIPTION
**Pull request by MingweiZhu666**
… OCI FireNet

### CODE REVIEW

File: aviatrix/utils.go [hunk 1]
```diff
@@ -298,6 +299,11 @@ func checkPAVMVersionFormat(version string) string {
 
 // compareVersion compares two Semantic Versions
 func compareVersion(version1, version2 string) bool {
+	defer func() {
```
Comment (mlin-aviatrix) [inline aviatrix/utils.go]: You probably don't need to defer here. `version.NewVersion` returns an error as its second return value. Right now you have not assigned it to anything. You can do something like this:
```
v1, err := version.NewVersion(version1)
if err != nil {
  // Log if needed
  return false
}
```

**Review by mlin-aviatrix — CHANGES_REQUESTED** [review]

File: aviatrix/utils.go (hunk 1 above)
Comment (MingweiZhu666) [reply aviatrix/utils.go:302]: that cash error comes from the v1.GreaterThan(v2) function and I check that function doesn't have the error as its second return value.

Comment (MingweiZhu666) [reply aviatrix/utils.go]: have fixed

File: aviatrix/utils.go [hunk 2]
```diff
@@ -298,8 +299,16 @@ func checkPAVMVersionFormat(version string) string {
 
 // compareVersion compares two Semantic Versions
 func compareVersion(version1, version2 string) bool {
-	v1, _ := version.NewVersion(version1)
-	v2, _ := version.NewVersion(version2)
+	v1, err := version.NewVersion(version1)
+	if err != nil {
+		log.Println("not support sort this version format: ", version1, err)
```
Comment (mlin-aviatrix) [inline aviatrix/utils.go]: This error message should be reworded and should probably use `log.Printf`. Something like `log.Printf("unsupported firewall image version format: %s\n", version1)`

**Review by mlin-aviatrix — CHANGES_REQUESTED** [review]

File: aviatrix/utils.go (hunk 2 above)
Comment (MingweiZhu666) [reply aviatrix/utils.go]: thank you, have fixed

**Review by mlin-aviatrix — APPROVED** [review]

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**